Tests/BotTests/GitHub/GitHubEventsTests.swift

```diff
@@ -2,33 +2,47 @@ import XCTest
 import Nimble
 import Result
 import ReactiveSwift
+import CryptoSwift
 @testable import Bot
 
 class GitHubEventsTests: XCTestCase {
 
+    let signatureToken = "Wall-E"
+
+    private func signature(for data: Data) -> String {
+        let signature = try! HMAC(key: signatureToken, variant: .sha1)
+            .authenticate(data.bytes)
+            .toHexString()
+
+        return "sha1=\(signature)"
+    }
+
     func test_handling_pull_request_event() {
-        let service = GitHubService()
+        let service = GitHubService(signatureToken: signatureToken)
         let scheduler = TestScheduler()
 
         let pullRequestEvent = PullRequestEvent(
-            action: .opened,
+            action: .closed,
             pullRequestMetadata: PullRequestMetadata(
                 reference: PullRequest(
                     number: 1,
-                    title: "Hello World",
-                    author: PullRequest.Author(login: "Wall-E Bot"),
-                    source: PullRequest.Branch(ref: "some-feature", sha: "123"),
-                    target: PullRequest.Branch(ref: "master", sha: "123"),
-                    labels: []
+                    title: "Update the README with new information",
+                    author: PullRequest.Author(login: "Codertocat"),
+                    source: PullRequest.Branch(ref: "changes", sha: "34c5c7793cb3b279e22454cb6750c80560547b3a"),
+                    target: PullRequest.Branch(ref: "master", sha: "a10867b14bb761a232cd80139fbd4c0d33264240"),
+                    labels: [.init(name: "bug")]
                 ),
                 isMerged: false,
                 mergeState: .clean
             )
         )
 
         let request = StubbedRequest(
-            headers: [GitHubService.HTTPHeader.event.rawValue: GitHubService.APIEvent.pullRequest.rawValue],
-            body: pullRequestEvent
+            headers: [
+                GitHubService.HTTPHeader.event.rawValue: GitHubService.APIEvent.pullRequest.rawValue,
+                GitHubService.HTTPHeader.signature.rawValue: signature(for: GitHubPullRequestEvent.data(using: .utf8)!)
```

Comment:
Thinking out loud: doesn't it defeat the purpose to call the method computing the `signature(for:)` dynamically as a fixture for the test? I mean, it feels like checking if `func f(_ x: Int) { x * 2 }` works by testing `expect(f(2)) == f(2)` instead of testing `expect(f(2)) == 4`, no?

Maybe it's more resilient to put the hardcoded expected signature there?

Reply:
I think you have a point here Olivier 👍